tools/generator/Java.Interop.Tools.Generator.Importers/CecilApiImporter.cs

```diff
@@ -249,11 +249,17 @@ static CustomAttribute GetObsoleteAttribute (Collection<CustomAttribute> attribu
 		static string GetObsoleteComment (CustomAttribute attribute) =>
 			attribute?.ConstructorArguments.Any () == true ? (string) attribute.ConstructorArguments [0].Value : null;
 
-		static CustomAttribute GetRegisterAttribute (Collection<CustomAttribute> attributes) =>
+		static CustomAttribute GetRegisterAttribute (Collection<CustomAttribute> attributes, CodeGenerationOptions opt) =>
```

Comment:
Shouldn't this overload instead take `ApiImporterOptions` so that it can check `ApiImporterOptions. SupportedRegisterAttributes`, a'la `GetRegisterAttribute()` within `ManagedApiImporter.cs`?

Reply:
The issue is that we're working in 2 different assemblies with both of these seemingly identical changes.

In `generator`, we have plumbed `CodeGenerationOptions` through most of the assembly, including the main public entry points to `CecilApiImporter`: `CreateClass` and `CreateInterface`.  In this case it makes the most sense to continue using our exist "options" class and plumb it a little deeper.

`ApiImporterOptions` is in `Java.Interop.Tools.JavaTypeSystem`, which did not previously have an "options" class so one had to be added.  As `JavaTypeSystem` is a more general assembly it should not have the concept of `JavaInterop1`/`XAJavaInterop1`, so the shape of the "options" class is a little different.